misk/src/main/kotlin/misk/inject/Guice.kt

```diff
@@ -37,6 +43,25 @@ internal class ListProvider<T> : Provider<List<T>> {
     override fun get(): List<T> = ArrayList(set)
 }
 
+internal inline fun <reified T : Any> subtypeOf(): WildcardType {
+    return Types.subtypeOf(T::class.java)
+}
+
+/**
+ * Returns the [TypeLiteral] of a [ParameterizedType], type parameterized on [typeParameters]
+ */
+internal inline fun <reified T : Any> parameterizedTypeLiteral(vararg typeParameters: Type): TypeLiteral<*> {
```

Comment:
I don’t think I like functions that accept _Type_ and return _TypeLiteral_ because they don’t compose

Reply:
easier to stay in one abstraction or the other

Reply:
```
parameterizedType(...).typeLiteral()
```

Reply:
Yeah I wasn't thrilled with it, moving back.